src/snapred/backend/recipe/algorithm/DetectorPeakPredictor.py

```diff
@@ -91,10 +99,16 @@ def PyExec(self) -> None:
                 widthLeft = fwhm * self.FWHMMultiplierLeft
                 widthRight = fwhm * self.FWHMMultiplierRight + self.peakTailCoefficient / beta_d
 
+                if self.purgeDuplicates:
+                    d = round(d, 5)
+
                 singleFocusGroupPeaks.append(
                     DetectorPeak(position=LimitedValue(value=d, minimum=d - widthLeft, maximum=d + widthRight))
                 )
 
+            if self.purgeDuplicates:
+                singleFocusGroupPeaks = list({peak.position.value: peak for peak in singleFocusGroupPeaks}.values())
+
```

Comment:
why not using a set, instead of a dictionary?

Reply:
Pydantic objects give an unhashable type exception.  I could introduce a hash method or do a pydantic "freeze" but this felt least like introducing a new moving part.

Reply:
No, the dictionary is fine